### DISCUSSION TITLE
Im using a dock able probe, cant get it to work with that config

### DISCUSSION THREAD

**Discussion by SvenixS** [jschuh/klipper-macros#151] [Q&A]
Im using Klack Ender probe which is dockable and when the print starts on the probing sequence it fails to deploy the probe and i cant find a way to add a wrapper for it

**Comment by Dissfall** [comment]
Hi! I've found a possible solution. It's seems ugly but works for me.

Let me know if it works or does not works for you.

## TL;DR

Add following configuration into your `printer.cfg`:

```
[gcode_macro list_macros]
description: Lists registered macros (and optionally show variable state).
  Usage: LISTVARS SEARCH=<search_string> VARS=<0|1> SILENT=<0|1>
variable_macros: {}
gcode:
  # Load the config state once and save it.
  {% if not macros %}
    {% set renames = {} %}
    {% for k in printer.configfile.config %}
      {% if k.lower().startswith("gcode_macro") and
            printer.configfile.config[k].rename_existing and not k.lower().endswith("bed_mesh_calibrate") %}
          {% set name = k.split(None, 1)[1] | lower %}
          {% set dummy = renames.__setitem__(name,
                          [printer.configfile.config[k].rename_existing|lower] +
                            renames[name]|default([])) %}
      {% endif %}
    {% endfor %}
    {% for k in printer %}
      {% if k.lower().startswith("gcode_macro") %}
        {% set name = k.split(None, 1)[1] | lower %}
        {% set dummy = macros.__setitem__(name, renames[name]|default([])) %}
      {% endif %}
    {% endfor %}
  {% endif %}

  {% if params.SILENT|default(0)|int == 0 %}
    _LIST_MACROS_DISPLAY {rawparams}
  {% endif %}

[gcode_macro BED_MESH_CALIBRATE]
rename_existing: BED_MESH_CALIBRATE_1
gcode:
  PROBE_OUT
  BED_MESH_CALIBRATE_1
  PROBE_IN
```

## Explanation

Klipper-macros by default uses macro called `BED_MESH_CALIBRATE_FAST` that uses build-in macro `BED_MESH_CALIBRATE`.

If we have a look at the definition of the `BED_MESH_CALIBRATE_FAST` we can see that it uses utility macro `list_macros` that lists all macros and it's names if they was renamed. This way `BED_MESH_CALIBRATE_FAST` uses only original definition of `BED_MESH_CALIBRATE`. You can even see comment about this:

https://github.com/jschuh/klipper-macros/blob/01c0d8aa06b39576501faaf9c905fae37d8aaef0/bed_mesh_fast.cfg#L15-L18

Thus if we will try to redefine `BED_MESH_CALIBRATE` and add our `PROBE_OUT` and `PROBE_IN `, `BED_MESH_CALIBRATE` will use original definition anyways.

```
[gcode_macro BED_MESH_CALIBRATE]
rename_existing: BED_MESH_CALIBRATE_1
gcode:
  PROBE_OUT
  BED_MESH_CALIBRATE_1
  PROBE_IN
```

So the only way I've found to change this behavior is to redefine `list_macros` and make it ignore that `BED_MESH_CALIBRATE` was renamed.

To be able to do that we should change the conditional statement here:
https://github.com/jschuh/klipper-macros/blob/01c0d8aa06b39576501faaf9c905fae37d8aaef0/globals.cfg#L425-L426

And make it look like that:
```
            printer.configfile.config[k].rename_existing and not k.lower().endswith("bed_mesh_calibrate") %}
          {% set name = k.split(None, 1)[1] | lower %}
```

Full declaration of the macro you can see in the beginning.
